drracket-tool-lib/drracket/private/standalone-module-browser.rkt

```diff
@@ -368,132 +362,128 @@
+           (label (string-constant module-browser-name-length))
+           (choices (list (string-constant module-browser-name-long)
+                          (string-constant module-browser-name-very-long)))
+           (selection (case (preferences:get 'drracket:module-browser:name-length)
+                        [(0) 0]
+                        [(1) 0]
+                        [(2) 0]
+                        [(3) 1]))
+           (callback
+            (λ (x y)
+              ;; note: the preference drracket:module-browser:name-length is also used for
+              ;; the View|Show Module Browser version of the module browser
+              ;; here we just treat any pref value except '3' as if it were for the long names.
+              (let ([selection (send module-browser-name-length-choice get-selection)])
+                (preferences:set 'drracket:module-browser:name-length (+ 2 selection))
+                (send pasteboard
+                      set-name-length
+                      (case selection
+                        [(0) 'long]
+                        [(1) 'very-long])))))))
+    (send pkg-choice set-string-selection (send pasteboard get-main-file-pkg))
+    
+    (define ec (make-object overview-editor-canvas% vp pasteboard))
+    
+    (define search-hp (new horizontal-panel% [parent vp] [stretchable-height #f]))
+    (define search-tf
+      (new text-field%
+           [label (string-constant module-browser-highlight)]
+           [parent search-hp]
+           [callback
+            (λ (tf evt)
+              (define val (send tf get-value))
+              (define reg (and (not (string=? val "")) (regexp (regexp-quote (send tf get-value)))))
+              (update-found-and-search-hits reg))]))
+    (define search-hits (new message% [parent search-hp] [label ""] [auto-resize #t]))
+    (define (update-found-and-search-hits reg)
+      (send pasteboard begin-edit-sequence)
+      (define count 0)
+      (define phases (set))
+      (let loop ([snip (send pasteboard find-first-snip)])
+        (when snip
+          (when (is-a? snip boxed-word-snip<%>)
+            (define fn (send snip get-filename))
+            (define found? (and reg fn (regexp-match reg (path->string fn))))
+            (when (or (not reg) found?)
+              (for ([phase (in-list (send snip get-require-phases))])
+                (set! phases (set-add phases phase)))
+              (set! count (+ count 1)))
+            (send snip set-found! found?))
+          (loop (send snip next))))
+    
+      (send search-hits
+            set-label
+            (string-append (if reg
+                               (format "~a found" count)
+                               (format "~a total" count))
+                           (render-phases phases)))
+      (send pasteboard end-edit-sequence))
+    (update-found-and-search-hits #f) ;; only to initialize search-hits
+    
+    (set! update-label
+          (λ (s)
+            (if (and s (not (null? s)))
```

Comment:
this should have become a `cond`?

Reply:
The `let*` expression in the true branch can't be refactored into definitions because of the shadowing it's doing in the binding for `lines`, so Resyntax won't refactor this `if` into a `cond`. It only does that for `if`s with `let`s when the `let` is refactorable.

Reply:
I'm not seeing a shadowing binding for `lines`. When I put `lines` just behind the `λ`, I get a free id error.

Reply:
Oh drat I misread this bit below:

<img width="508" alt="Screenshot 2024-12-01 at 11 21 12 AM" src="https://github.com/user-attachments/assets/ee08cbe2-f9b4-4536-88be-373ebbcff0aa">

I checked out the code and ran Resyntax over it manually and it did indeed fix this code, so I think the issue is it just ran up against the fix limit.